specs/core/1_shard-data-chains.md

```diff
@@ -0,0 +1,122 @@
+# Ethereum 2.0 Phase 1 -- Shard Data Chains
+
+###### tags: `spec`, `eth2.0`, `casper`, `sharding`
+
+**NOTICE**: This document is a work-in-progress for researchers and implementers. It reflects recent spec changes and takes precedence over the [Python proof-of-concept implementation](https://github.com/ethereum/beacon_chain).
+
+### Introduction
+
+This document represents the specification for Phase 1 of Ethereum 2.0 -- Shard Data Chains. Phase 1 depends on the implementation of [Phase 0 -- The Beacon Chain](0_beacon-chain.md).
+
+Ethereum 2.0 consists of a central beacon chain along with `SHARD_COUNT` shard chains. Phase 1 is primarily concerned with the construction, validity, and consensus on the _data_ of these shard chains. Phase 1 does not specify shard chain state execution or account balances. This is left for future phases.
+
+### Terminology
+
+### Constants
+
+Phase 1 depends upon all of the constants defined in [Phase 0](0_beacon-chain.md#constants) in addition to the following:
+
+| Constant | Value | Unit | Approximation |
+| `CHUNK_SIZE` | 2**8 (= 256) | bytes |
+| `MAX_SHARD_BLOCK_SIZE` | 2**15 (= 32768) | bytes |
+
+## Data Structures
+
+### Shard chain blocks
+
+A `ShardBlock` object has the following fields:
+
+```python
+{
+    # Slot number
+    'slot': 'uint64',
+    # Parent block hash
+    'parent_hash': 'hash32',
+    # Beacon chain block
+    'beacon_chain_ref': 'hash32',
+    # Depth of the Merkle tree
+    'data_tree_depth': 'uint8',
+    # Merkle root of data
+    'data_root': 'hash32'
+    # State root (placeholder for now)
+    'state_root': 'hash32',
+    # Attestation (including block signature)
+    'attester_bitfield': 'bytes',
+    'aggregate_sig': ['uint256'],
+}
+```
+
+## Shard block processing
+
+A block on a shard can be processed only if its `parent` has been accepted. To validate a block header on shard `shard_id`, compute as follows:
+
+* Verify that `beacon_chain_ref` is the hash of the `slot`'th block in the beacon chain.
+* Let `state` be the state of the beacon chain block referred to by `beacon_chain_ref`. Let `validators` be `[validators[i] for i in state.current_persistent_committees[shard_id]]`.
+* Assert `len(attester_bitfield) == ceil_div8(len(validators))`
+* Let `curblock_proposer_index = hash(state.randao_mix + bytes8(shard_id)) % len(validators)`. Let `parent_proposer_index` be the same value calculated for the parent block.
+* Make sure that the `parent_proposer_index`'th bit in the `attester_bitfield` is set to 1.
+* Generate the group public key by adding the public keys of all the validators for whom the corresponding position in the bitfield is set to 1. Verify the `aggregate_sig` using this as the pubkey and the `parent_hash` as the message.
+
+Note that at network layer we expect blocks to be broadcasted along with the signature from the `curblock_proposer_index`'th validator in the validator set for that block.
```

Comment:
Move this to the top of the section and have it as one of the pre-processing conditions (alongside parent being accepted)